inc/class-config.php

```diff
@@ -0,0 +1,163 @@
+<?php
+/**
+ * Core platform config loader.
+ *
+ * @package hm-platform-core
+ */
+
+namespace HM\Platform;
+
+class Config {
+
+	/**
+	 * Primary config array.
+	 *
+	 * @var array
+	 */
+	protected static $config = [];
+
+	/**
+	 * Retrieve the configuration for HM Platform.
+	 *
+	 * The configuration is defined by merging the defaults set by modules
+	 * with any overrides present in composer.json.
+	 *
+	 * @return array Configuration data.
+	 */
+	static function get() : array {
+		if ( self::$config ) {
+			return self::$config;
+		}
+
+		self::$config = self::get_merged_defaults_and_customizations();
+
+		return self::$config;
+	}
+
+	/**
+	 * The environment type is used to override the config
+	 * as well as allow for checks within a project codebase.
+	 *
+	 * Defaults to local, this constant will be set at the stack level.
+	 *
+	 * @return string The environment type string.
+	 */
+	static function get_environment_type() : string {
+		if ( defined( 'HM_ENV_TYPE' ) ) {
+			return HM_ENV_TYPE;
+		}
+
+		return 'local';
+	}
+
+	/**
+	 * Merge the defaults and the contents of the various configuration files into a single configuration.
+	 *
+	 * @return array Configuration data.
+	 */
+	protected static function get_merged_defaults_and_customizations() : array {
+		// Default config.
+		$config = [
+			'modules' => [],
+			'environments' => [
+				'local' => [],
+				'development' => [],
+				'staging' => [],
+				'production' => [],
+			],
+		];
+
+		// @todo get registered modules & their defaults here.
+
+		if ( ! defined( 'ROOT_DIR' ) ) {
```

Comment:
We should set this in the `wp-config.php` of CMS probably, so we don't need to check for it.

Reply:
Added that to the `cms` repo PR 👍